src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -357,49 +363,84 @@ impl OpenIDConnectAuthProvider {
+                                                    r.to_string()
+                                                )));
+                                            }
+                                            openidconnect::RequestTokenError::Other(err_string) => {
+                                                match err_string.as_str() {
+                                                    "temporarily_unavailable" | "server_error" => {
+                                                        return Err(CoreErrorResponseType::Extension(
+                                                            err_string.to_string(),
+                                                        ))
+                                                    }
+                                                    _ => {
+                                                        return Err(CoreErrorResponseType::Extension(format!(
+                                                            "Unknown Error(#1) while receiving new token: {}",
+                                                            err_string
+                                                        )))
+                                                    }
+                                                }
+                                            }
+                                        }
                                     }
                                 }
                             }
                             None => {
                                 // should be unreachable
-                                warn!(
-                                    "OpenID Connect: Could not extend login session for user '{}': {}",
-                                    &session.id, "Connection to OpenID Connect provider not yet established"
-                                );
+                                return Err(CoreErrorResponseType::Extension(
+                                    "Connection to OpenID Connect provider not yet established".to_string(),
+                                ));
                             }
                         }
                     }
                     Err(err) => {
-                        warn!(
-                            "OpenID Connect: Could not extend login session for user '{}': {}",
-                            &session.id, err
-                        );
+                        return Err(CoreErrorResponseType::Extension(
+                            format!("Unknown Error(#2) while receiving new token: {}", err).to_string(),
+                        ));
                     }
                 }
             }
             None => {
-                debug!(
-                    "OpenID Connect: Could not extend login session for user '{}': no refresh token available",
-                    &session.id
-                );
+                return Err(CoreErrorResponseType::Extension(
```

Comment:
As discussed move this case to the caller, i.e. don't try to refresh if we don't have a refresh token rather than attempting to encode this as an error.